core/templates/dev/head/app.js

```diff
@@ -270,6 +203,186 @@ oppia.config(['$provide', function($provide) {
   ]);
 }]);
 
+// Dynamically generate CKEditor widgets for the rich text components.
+oppia.run([
+  '$timeout', '$compile', '$rootScope', '$uibModal', 'RteHelperService',
+  'HtmlEscaperService', 'taOptions',
+  function($timeout, $compile, $rootScope, $uibModal, RteHelperService,
+      HtmlEscaperService, taOptions) {
+    taOptions.disableSanitizer = true;
+    taOptions.forceTextAngularSanitize = false;
+    taOptions.classes.textEditor = 'form-control oppia-rte-content';
+    taOptions.setup.textEditorSetup = function($element) {
+      $timeout(function() {
+        $element.trigger('focus');
+      });
+    };
+    var _RICH_TEXT_COMPONENTS = RteHelperService.getRichTextComponents();
+    _RICH_TEXT_COMPONENTS.forEach(function(componentDefn) {
+      // The name of the CKEditor widget corresponding to this component.
+      var ckName = 'oppia' + componentDefn.id;
+
+      // For some reason, frontend tests will error without this check.
```

Comment:
Would be good to have a better explanation here.

Reply:
Done!